juju/testing/apiserver.go

```diff
@@ -412,6 +398,34 @@ func (s *ApiServerSuite) TearDownTest(c *gc.C) {
 	s.MgoSuite.TearDownTest(c)
 }
 
+// ServiceFactory returns a service factory for the model.
+func (s *ApiServerSuite) ServiceFactory(modelUUID string) servicefactory.ServiceFactory {
+	return s.ServiceFactoryGetter.FactoryForModel(modelUUID)
+}
+
+// SeedControllerCloud is responsible for applying the controller cloud to
+// the given database.
+func (s *ApiServerSuite) SeedControllerCloud(c *gc.C, runner coredatabase.TxnRunner) {
+	err := InsertDummyCloudType(context.Background(), s.TxnRunner())
+	c.Assert(err, jc.ErrorIsNil)
+	err = cloudbootstrap.InsertInitialControllerCloud(DefaultCloud)(context.Background(), runner)
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+// AllowCloudType is responsible for applying the cloud type to
+// the given database.
+func AllowCloudType(ctx context.Context, db database.TxnRunner, version int, name string) error {
```

Comment:
Should we move this to the cloud state? It's something that we may or may not need some where download the line?